framework/logging.cpp

```diff
@@ -2483,17 +2485,20 @@ void YamlLogger::print()
     }
 
     // print the thread messages
-    auto doprint = [this](PerThreadData::Common *data, int i) {
+    auto doprint = [this, init_skip_message_bytes](PerThreadData::Common *data, int s_tid) {
         struct mmap_region r = maybe_mmap_log(data);
 
-        if (r.size == 0 && !data->has_failed() && sApp->shmem->verbosity < 3)
+        /* for main threads (negative ids) adjust the message size to account
+         * for skip message. this is to avoid empty messages printed to the log.
+         */
+        if (r.size - (s_tid < 0 ? init_skip_message_bytes : 0) == 0 && !data->has_failed() && sApp->shmem->verbosity < 3)
```

Comment:
Keep the current line as-is, because if `r.size` is zero, there definitely isn't anything to send. Then add:
```c++
/* explaining comment here */
if (s_tid < 0 && r.size <= init_skip_message_bytes)
    return;
```

This will also prevent a problem in the case `init_skip_message_bytes > r.size` (see above) that would result in a non-zero value here.

Reply:
It absolutely must print stuff that is not a skip message, right? Skip message will not be printed if the test was skipped, but all the rest will be. So to me the logic seems correct.